src/components/EvolutionTab.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from "react";
+import { fetchPokemon } from "../util/fetchPokemonData";
+
+const EvolutionTab = ({ evolutionChainUrl }) => {
+  const [evolutionNamesObjects, setEvolutionNamesObjects] = useState([]);
+  const [evolutionNameStrings, setEvolutionNameStrings] = useState([]);
+  const [spriteUrls, setSpriteUrls] = useState([]);
+
+  let evolutionNamesList = [];
+  let chain = {};
+
+  // Function to fetch chain object, save to state hook
+  const fetchChain = async () => {
+    try {
+      const chainData = await fetch(evolutionChainUrl)
+        .then((response) => response.json())
+        .then((data) => {
+          chain = data.chain;
+        });
+      evolutionNamesList = getEvolutionNames(chain);
+      setEvolutionNamesObjects(evolutionNamesList);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  // search through each pokemon object and get all names of evolution into array
+  const getEvolutionNames = (chainObj) => {
+    let evoChain = [];
+    let evoData = chainObj;
+
+    if (chainObj) {
+      do {
+        let numberOfEvolutions = evoData["evolves_to"].length;
+        evoChain.push({
+          species_name: evoData.species.name,
+        });
+
+        if (numberOfEvolutions > 1) {
+          for (let i = 1; i < numberOfEvolutions; i++) {
+            evoChain.push({
```

Comment:
I suggest checking out for of loops. Theyre much easier to read than these kinds of loops. 
Instead of for(let i = 1; i < numberOfEvolutions; i++) a for of loop would be `for(const evolutions of numberOfEvolutions { whatever you want to do for each evolution}`

Reply:
Good suggestion!  `forOf` would definitely be cleaner and more readable for this.